### PR TITLE
[Glibc] Define 'extern char **environment' correctly on FreeBSD.

### PR DESCRIPTION
**Pull request by dcci**

### CODE REVIEW

File: stdlib/public/Glibc/Glibc.swift [hunk 1]
```diff
@@ -219,3 +219,14 @@ public func sem_open(
...
+
+@warn_unused_result
+@_silgen_name("_swift_FreeBSD_getEnv")
+func _swift_FreeBSD_getEnv(
+) -> UnsafeMutablePointer<UnsafeMutablePointer<UnsafeMutablePointer<CChar>>>
+
+@warn_unused_result
+public func _FreeBSDGetEnviron(
+) -> UnsafeMutablePointer<UnsafeMutablePointer<UnsafeMutablePointer<CChar>>> {
+  return _swift_FreeBSD_getEnv();
```
Comment (gribozavr) [inline stdlib/public/Glibc/Glibc.swift]: Extra semicolon?

File: stdlib/public/Glibc/Glibc.swift [hunk 2]
```diff
@@ -219,3 +219,14 @@ public func sem_open(
...
   return _swift_Glibc_sem_open4(name, oflag, mode, value)
 }
+
+@warn_unused_result
+@_silgen_name("_swift_FreeBSD_getEnv")
+func _swift_FreeBSD_getEnv(
+) -> UnsafeMutablePointer<UnsafeMutablePointer<UnsafeMutablePointer<CChar>>>
+
+@warn_unused_result
+public func _FreeBSDGetEnviron(
```
Comment (gribozavr) [inline stdlib/public/Glibc/Glibc.swift]: Could you instead add an overlay for `var environ`, for everyone's benefit?

Comment (dcci) [inline stdlib/public/Glibc/Glibc.swift]: hmm, you mean like we do for errno in the very same file, right?

Comment (gribozavr) [inline stdlib/public/Glibc/Glibc.swift]: Right.

**Comment by dcci** [comment]
__environment is now a variable, as per Dmitri request.

File: stdlib/public/Glibc/Glibc.swift [hunk 3]
```diff
@@ -219,3 +219,17 @@ public func sem_open(
...
+
+// FreeBSD defines extern char **enviroment differently than Linux.
+#if os(FreeBSD)
+@warn_unused_result
+@_silgen_name("_swift_FreeBSD_getEnv")
+func _swift_FreeBSD_getEnv(
+) -> UnsafeMutablePointer<UnsafeMutablePointer<UnsafeMutablePointer<CChar>>>
+
+public var __environ: UnsafeMutablePointer<UnsafeMutablePointer<CChar>> {
+  get {
```
Comment (gribozavr) [inline stdlib/public/Glibc/Glibc.swift]: You can drop `get {` in this case.

**Comment by gribozavr** [comment]
@dcci Why did you double-underscore prefix it?  If it is an API on FreeBSD without the underscores, it should be exposed like that in Swift, I think.

**Comment by dcci** [comment]
Addressed Dmitri's comments, round 2.

**Pull request merged**